pyproject.toml

```diff
@@ -83,11 +83,11 @@ filterwarnings = [
     'ignore:the `interpolation=` argument to percentile was renamed to `method=`, which has additional options:DeprecationWarning',  # Issue #1772
     "ignore:The interpolation= argument to 'quantile' is deprecated. Use 'method=' instead:DeprecationWarning",  # Issue #1772
     'ignore: Exception ignored in:pytest.PytestUnraisableExceptionWarning',  #FIXME: Exception ignored in: <_io.FileIO [closed]>
-    'ignore:invalid value encountered in true_divide:RuntimeWarning',  #FIXME
+    'ignore:invalid value encountered in (?:true_)?divide:RuntimeWarning',  #FIXME
```

Comment:
@kratsg hm. [`warnings.filterwarnings`](https://docs.python.org/3/library/warnings.html#warnings.filterwarnings) doesn't like either of our regexs

```
AttributeError: module 'builtins' has no attribute 'true_)?divide'
```

Both of ours were valid by themselves, but they don't seem to want to match.

Reply:
just copy/paste and do two identical lines... seems to be a terrible regex parser somehow.

Reply:
```python
import warnings

message_true_divide = "invalid value encountered in true_divide"
message_divide = "invalid value encountered in divide"

message_ignore_regex = "invalid value encountered in (?:true_)?divide"

warnings.filterwarnings("ignore", message=message_ignore_regex, category=Warning)

warnings.warn(message_true_divide)
warnings.warn(message_divide)
```

works as expected, so it must be something with how [pytest is doing `filterwarnings`](https://docs.pytest.org/en/7.1.x/how-to/capture-warnings.html).

This would be good to understand for the future, but this PR is preventative anyway so this can get ignored for a bit. The only reason I opened it up tonight is that I thought it would be a 10 minute thing before bed &mdash; seems I was wrong.